src/main/java/seedu/address/logic/parser/DeleteTagCommandParser.java

```diff
@@ -10,21 +12,36 @@
 /**
  * Parses input arguments and creates a new DeleteTagCommand object
  */
-public class DeleteTagCommandParser {
+public class DeleteTagCommandParser implements Parser<DeleteTagCommand> {
 
     /**
      * Parses the given {@code String} of arguments in the context of the DeleteTagCommand
      * and returns an DeleteTagCommand object for execution.
      * @throws ParseException if the user input does not conform the expected format
      */
     public DeleteTagCommand parse(String args) throws ParseException {
-        try {
-            Tag tag = new Tag(args);
-            return new DeleteTagCommand(tag);
-        } catch (IllegalValueException ive) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, DeleteTagCommand.MESSAGE_USAGE));
+        requireNonNull(args);
```

Comment:
requireNonNull throws a NullPointerException which is not handled. The parser should throw a ParseException instead.